kernel-open/common/inc/nv.h

```diff
@@ -180,9 +180,9 @@ extern const char *pNVRM_ID;
 
 typedef union
 {
-    volatile NvV8 Reg008[1];
-    volatile NvV16 Reg016[1];
-    volatile NvV32 Reg032[1];
+    volatile NvV8 Reg008[2564];
```

Comment:
This doesn't seem right. How is changing the array sizes this drastically an appropriate fix for a compiler warning? Even if the pointer typedef (nv_phwreg_t) is meant to be a convenience type for register address casts how can you be sure that the non-pointer type (nv_hwreg_t) isn't used by some external codebase in a way that depends on the size? And is the new size guaranteed to be 2564 bytes everywhere and always? If it is then why use a magic number instead of a macro with an informative name?

Reply:
> how can you be sure that the non-pointer type (nv_hwreg_t) isn't used by some external codebase in a way that depends on the size?

I highly doubt it's anything more than a convenience type. It wouldn't make sense for each field to have a length of 1 otherwise.

> And is the new size guaranteed to be 2564 bytes everywhere and always - and even if it is, why use a magic number instead of a macro with an informative name?

That's very fair, I would've hoped a potential reviewer could show a better way of doing things ;) Either way, I don't think this is worse than leaving a 1 and cluttering output with warnings which can easily be silenced.